### PR TITLE
Validate TextIOWrapper chunk size

### PR DESCRIPTION
**Pull request by shAn-kor**
- [x] This PR follows our [AI policy](https://github.com/RustPython/.github/blob/main/AI_POLICY.md)

## Summary

Makes the `TextIOWrapper._CHUNK_SIZE` setter match CPython's validation behavior.

The setter now:

- accepts integers and objects implementing `__index__()`
- rejects values less than or equal to zero with `ValueError`
- rejects non-indexable values with `TypeError`
- rejects values outside the platform `isize` range with `ValueError`
- reports the original object's type name for oversized indexable values
- checks uninitialized and detached wrappers before conversion
- preserves the existing deletion error behavior

The regression snippet covers boundary values, indexable objects, oversized results, uninitialized and detached wrappers, deletion, and error-message truncation.

## Tests

```shell
.git/pre-push-venv/bin/prek run --files extra_tests/snippets/stdlib_io.py

cd extra_tests
RUSTPYTHON=../target/debug/rustpython ../.git/pre-push-venv/bin/pytest -v test_snippets.py -k stdlib_io
```

Result: `6 passed, 392 deselected`.

The snippet was also run directly with CPython and RustPython, both normally and with `-O`.

## AI assistance

Codex (GPT-5) was used to compare RustPython with CPython, help draft the Rust implementation and regression tests, and help address review feedback. I reviewed the final diff and verified the behavior locally.

### CODE REVIEW

**Review by ShaharNaveh — APPROVED** [review]
tysm:)

welcome to the project

File: extra_tests/snippets/stdlib_io.py [hunk 1]
```diff
@@ -84,3 +84,30 @@ def write(self, data):
 raw.textio = textio
 with assert_raises(AttributeError):
     textio.writelines(["x"])
+
+textio = TextIOWrapper(BytesIO())
```
Comment (youknowone) [inline extra_tests/snippets/stdlib_io.py:88]: was there tests in `Lib/test/test_io.py`? we usually don't add test in extra_tests when we already have one in Python unit test.
This is okay if there is no test hit this behavior

**Review by youknowone — CHANGES_REQUESTED** [review]

File: extra_tests/snippets/stdlib_io.py (hunk 1 above)
Comment (shAn-kor) [reply extra_tests/snippets/stdlib_io.py:88]: I checked `test_io`. It has a few `_CHUNK_SIZE` tests, but none of them cover the validation changed here. I also couldn't find boundary-value coverage for this setter in CPython's current test suite.

File: crates/vm/src/stdlib/_io.rs [hunk 2]
```diff
@@ -3206,19 +3206,27 @@ mod _io {
...
-            let mut textio = self.lock(vm)?;
-            match chunk_size {
-                PySetterValue::Assign(chunk_size) => textio.chunk_size = chunk_size,
+        fn set_chunksize(&self, value: PySetterValue, vm: &VirtualMachine) -> PyResult<()> {
+            let chunk_size: isize = match value {
+                PySetterValue::Assign(object_value) => {
+                    let integer = object_value.try_index(vm)?;
+
+                    integer.try_to_primitive::<isize>(vm).map_err(|_| {
+                        vm.new_value_error("cannot fit 'int' into an index-sized integer")
```
Comment (ShaharNaveh) [inline crates/vm/src/stdlib/_io.rs]: Is it always "int"? don't we need to take `object_value.class().name()` instead?

**Review by ShaharNaveh — CHANGES_REQUESTED** [review]

File: crates/vm/src/stdlib/_io.rs (hunk 2 above)
Comment (shAn-kor) [reply crates/vm/src/stdlib/_io.rs]: Thanks for catching that. CPython uses the original object's type name here, not always int. I'll update the error message and add a test for an indexable object that returns an oversized integer.

File: extra_tests/snippets/stdlib_io.py [hunk 3]
```diff
@@ -84,3 +84,99 @@ def write(self, data):
...
+
+uninitialized_textio = TextIOWrapper.__new__(TextIOWrapper)
+uninitialized_chunk_size = UninitializedChunkSize()
+expect_value_error(
+    "I/O operation on uninitialized object",
+    lambda: setattr(uninitialized_textio, "_CHUNK_SIZE", uninitialized_chunk_size),
+)
+
+if uninitialized_chunk_size.called:
+    raise AssertionError("__index__ should not be called for uninitialized TextIOWrapper")
```
Comment (fanninpm) [inline extra_tests/snippets/stdlib_io.py]: `ruff` is complaining about the formatting of this line.

**Comment by fanninpm** [comment]
If the CI failure bothers you, you can rebase this branch onto a fresh copy of `main`.

**Review by ShaharNaveh — APPROVED** [review]
tysm!

**Comment by shAn-kor** [comment]
> If the CI failure bothers you, you can rebase this branch onto a fresh copy of `main`.

Oh, Thank you so much!

**Comment by fanninpm** [comment]
> > If the CI failure bothers you, you can rebase this branch onto a fresh copy of `main`.
> 
> Oh, Thank you so much!

Keep in mind that this only worked because #8291 had just been merged. I didn't mention that.

**Review by youknowone — APPROVED** [review]
👍
